library/src/updater.rs

```diff
@@ -437,6 +456,19 @@ pub fn update() -> UpdateStatus {
     });
 }
 
+/// This does not return status.  The only output is the change to the saved
+/// cache.  The Engine calls this during boot and it will check for an update
+/// and install it if available.
+pub fn start_update_thread() {
+    // This holds the lock on the config for the entire duration of the update
```

Comment:
Is there any harm in holding the lock on the config?

Reply:
There is, in that any other calls into the library will block until the lock is released. Which means that if we add a dart api those calls would hang while an update is in progress.  Better is to have it drop the lock while it's doing network requests since those can go on for a long time.